classes/Task/Update/Download.php

```diff
@@ -69,30 +70,39 @@ public function run(): int
         $report = '';
         $relative_download_path = str_replace(_PS_ROOT_DIR_, '', $this->container->getProperty(UpgradeContainer::DOWNLOAD_PATH));
         if (\ConfigurationTest::test_dir($relative_download_path, false, $report)) {
-            $res = $upgrader->downloadLast(
-                $this->container->getProperty(UpgradeContainer::DOWNLOAD_PATH),
-                $this->container->getProperty(UpgradeContainer::ARCHIVE_FILENAME)
-            );
-            if ($res) {
-                $md5file = md5_file(realpath($this->container->getProperty(UpgradeContainer::ARCHIVE_FILEPATH)));
-                if ($md5file == $upgrader->getOnlineDestinationRelease()->getZipMd5()) {
-                    $this->next = TaskName::TASK_UNZIP;
-                    $this->logger->debug($this->translator->trans('Download complete.'));
-                    $this->logger->info($this->translator->trans('Download complete. Now extracting...'));
-                } else {
-                    $this->logger->error($this->translator->trans('Download complete but MD5 sum does not match (%s).', [$md5file]));
-                    $this->logger->info($this->translator->trans('Download complete but MD5 sum does not match (%s). Operation aborted.', [$md5file]));
-                    $this->next = TaskName::TASK_ERROR;
-                }
-            } else {
-                $this->logger->error($this->translator->trans('Error during download'));
-                $this->next = TaskName::TASK_ERROR;
-            }
+            $this->downloadArchive();
         } else {
             $this->logger->error($this->translator->trans('Download directory %s is not writable.', [$this->container->getProperty(UpgradeContainer::DOWNLOAD_PATH)]));
             $this->next = TaskName::TASK_ERROR;
         }
 
         return $this->next == TaskName::TASK_ERROR ? ExitCode::FAIL : ExitCode::SUCCESS;
     }
+
+    /**
+     * @throws DistributionApiException
+     * @throws UpgradeException
+     * @throws Exception
+     */
+    public function downloadArchive(): void
+    {
+        $destPath = realpath($this->container->getProperty(UpgradeContainer::DOWNLOAD_PATH)) . DIRECTORY_SEPARATOR . $this->container->getProperty(UpgradeContainer::ARCHIVE_FILENAME);
+        $archiveUrl = $this->container->getUpgrader()->getOnlineDestinationRelease()->getZipDownloadUrl();
+        $res = $this->container->getDownloadService()->downloadWithRetry($destPath, $archiveUrl);
+        if ($res) {
+            $md5file = md5_file(realpath($this->container->getProperty(UpgradeContainer::ARCHIVE_FILEPATH)));
+            if ($md5file == $this->container->getUpgrader()->getOnlineDestinationRelease()->getZipMd5()) {
+                $this->next = TaskName::TASK_UNZIP;
+                $this->logger->debug($this->translator->trans('Download complete.'));
+                $this->logger->info($this->translator->trans('Download complete. Now extracting...'));
```

Comment:
Shall we take the opportunity to simplify the contents sent to the logger?
I guess it was necessary before because the info was in a separate panel from the rest of the logs, but as it is not the case anymore with the v7, the call to `info()` seems enough.